ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ConsulBasedTargetProvider.java

```diff
@@ -73,6 +75,10 @@ private int instanceWeight(final HealthInfoInstance.Service instance) {
     checkState(!currTargets.isEmpty(), "No targets are currently registered for module " + healthyTargetsList.getModule());
     checkArgument(targetsNum > 0, "targets number must be more than zero");
 
+    return provideTargetsImpl(targetsNum, currTargets);
+  }
+
+  protected List<String> provideTargetsImpl(int targetsNum, List<String> currTargets) {
```

Comment:
I really dislike this kind of extension by inheritance. If you want to allow easy and safe extension - add an injected strategy.